qutebrowser/browser/browsertab.py

```diff
@@ -133,12 +134,97 @@ class TabData:
     fullscreen = attr.ib(False)  # type: bool
     netrc_used = attr.ib(False)  # type: bool
     input_mode = attr.ib(usertypes.KeyMode.normal)  # type: usertypes.KeyMode
+    splitter = attr.ib(None)
 
     def should_show_icon(self) -> bool:
         return (config.val.tabs.favicons.show == 'always' or
                 config.val.tabs.favicons.show == 'pinned' and self.pinned)
 
 
+class InspectorSplitter(QSplitter):
+
+    """Allows putting an inspector inside the tab.
+
+    Attributes:
+        _main_idx: index of the main webview widget
+        _position: position of the inspector (right/left/top/bottom)
+        _preferred_size: the preferred size of the inpector widget in pixels
+    """
+
+    _main_idx = 0
+    _position = None
+    _preferred_size = None
+
+    def __init__(self, main_webview):
+        super().__init__()
+        self.addWidget(main_webview)
+        self.splitterMoved.connect(self._onSplitterMoved)
+
+    def set_inspector(self, inspector, position):
+        """Set the position of the inspector."""
+        assert position in ['right', 'left', 'top', 'bottom']
```

Comment:
Can you change these lists to sets instead, ie: `assert position in {'right', 'left', 'top', 'bottom'}` when comparing against strings with `in`?

Reply:
I would feel a lot more comfortable if position was an enum with string values. That way there's at least a chance for the type checker to catch issues.

Reply:
If I'm doing something like:
```
InspectorPosition = enum.Enum('InspectorPosition', ['window', 'right', 'left', 'top', 'bottom'])
```
would it be acceptable to use
```
InspectorPosition.__dict__[position]
```
where `position` is the user input argument in `CommandDispatcher.toggle_inspector()`?

Or what would be best practice to turn a user input string into an enum?

Reply:
https://stackoverflow.com/questions/41407414/convert-string-to-enum-in-python -> `InspectorPosition[position]`.